bench/mjolnir/buildtiles.cc

```diff
@@ -0,0 +1,41 @@
+#include <benchmark/benchmark.h>
+
+#include "baldr/rapidjson_utils.h"
+#include "mjolnir/util.h"
+
+using namespace valhalla::mjolnir;
+
+int main(int argc, char** argv) {
+  valhalla::midgard::logging::Configure({{"type", ""}});
+
+  const auto build_tiles = [](benchmark::State& state, const std::string& pbf_path,
+                              BuildStage stage) {
+    const std::string kConfigFilePath = VALHALLA_SOURCE_DIR "bench/mjolnir/config.json";
+    boost::property_tree::ptree config;
+    rapidjson::read_json(kConfigFilePath, config);
+    config.get_child("mjolnir").erase("tile_extract");
+    config.get_child("mjolnir").erase("tile_url");
+
+    for (auto _ : state) {
+      state.PauseTiming();
+      build_tile_set(config, {pbf_path}, BuildStage::kInitialize,
+                     static_cast<BuildStage>(static_cast<uint8_t>(stage) - 1));
+      state.ResumeTiming();
+      benchmark::DoNotOptimize(build_tile_set(config, {pbf_path}, stage, stage));
+    }
+  };
+
+  const std::string kProtobufs[] = {"stockholm.osm.pbf"};
```

Comment:
Turned out that end-to-end building takes quite a lot time so bench runs only 1 iteration even with small pbfs(like this one)

Reply:
I don't mind merging such benchmarks to the master. However, I have doubts they'll be useful enough. A few thoughts of mine
1) Building tiles from small pbfs & large pbfs give 2 different times. E.g a stage that takes quite a lot time with small input data might work pretty fast on large amount of data. So we still need to test our changes/feature branch on larger input
2) My intuition tells me that benchmarks should measure fast functions that may occur in "hot spots"(e.g there're a lot of benchmarks of `insert`/`erase` stl methods). On the other hand, our stage-functions are called once during tile-building. What might work here is to add hot-functions-benchmark later: need to collect perf/gperftools statistics with large pbf and choose functions/methods with the most spent time(%)
3) The same code measurements might deviate by +- 5-10% sometimes(somewhy) so you might not even notice speed up

Reply:
It's possible to tell gbench to run more iterations:
```
benchmark [--benchmark_list_tests={true|false}]
          [--benchmark_filter=<regex>]
          [--benchmark_min_time=<min_time>]
          [--benchmark_repetitions=<num_repetitions>]
          [--benchmark_report_aggregates_only={true|false}]
          [--benchmark_display_aggregates_only={true|false}]
          [--benchmark_format=<console|json|csv>]
          [--benchmark_out=<filename>]
          [--benchmark_out_format=<json|console|csv>]
          [--benchmark_color={auto|true|false}]
          [--benchmark_counters_tabular={true|false}]
          [--v=<verbosity>]
```

Reply:
Is it possible to choose a smaller pbf to build tiles from in order to speed it up as well?

Reply:
> It's possible to tell gbench to run more iterations, using `--benchmark_min_time` I think
> 
> ```
> benchmark [--benchmark_list_tests={true|false}]
>           [--benchmark_filter=<regex>]
>           [--benchmark_min_time=<min_time>]
>           [--benchmark_repetitions=<num_repetitions>]
>           [--benchmark_report_aggregates_only={true|false}]
>           [--benchmark_display_aggregates_only={true|false}]
>           [--benchmark_format=<console|json|csv>]
>           [--benchmark_out=<filename>]
>           [--benchmark_out_format=<json|console|csv>]
>           [--benchmark_color={auto|true|false}]
>           [--benchmark_counters_tabular={true|false}]
>           [--v=<verbosity>]
> ```

I found `Iteration` method to set fixed number of iterations. Looks ok from the first sight

Reply:
> Is it possible to choose a smaller pbf to build tiles from in order to speed it up as well?

Turned out `shortcuts` stage takes significant amount of time even with small pbf(300ms -> 30ms when shortcuts-opiton is disabled). Though with enabled shortucts-option it takes 300ms for the end-to-end build. Looks acceptable for the benchmark